### DISCUSSION TITLE
Authorization Pending error while getting the Access Token from API

### DISCUSSION THREAD

**Discussion by nncodestone** [fattureincloud/api#388] [Q&A] [answered]
Hi,

I am getting Authorization Pending error while getting the Access Token from API. Can you please enable  our App "ByD Tests", ID is 7860.

Company is  Big Bus

Please let me know if you need any further details.

Thanks & Regards
Nageswara

**Comment by valmoz** [comment]
Hi @nncodestone 
what are you trying to do?
If you're trying to use the OAuth2.0 Device Code flow to obtain an access token, then you must contact our Customer Support to be authorized; they will send you a few questions to see if it makes sense to accept your request, so please try to explain in detail why you need this particular flow and why the other two are not good for your use case.

Here you can find a list of useful links:
- [Device Code Flow](https://developers.fattureincloud.it/docs/authentication/device-code)
- [Which Authentication Flow should I use?](https://developers.fattureincloud.it/docs/basics/authentication)
- [Support and Feature Request](https://developers.fattureincloud.it/docs/support)

Please, keep in mind that in this period of the year, the customer support team will take more time than usual to give you a response.

Thanks
Mauro

**Comment by nncodestone** [comment]
Hi Mauro,


We would like to go for Manual authentication to get the access token, Unfortunately we are unable to get it by following the documentation.
May I know Fattureincloud Customer care  email id?

**Reply by valmoz** [reply]
Hi @nncodestone 
if you're trying to obtain a Manual Token, then you shouldn't receive that error message. We have three different authentication methods, but only the **OAuth 2.0 Device Code** needs authorization from our Customer Support team; the other two methods (**Manual Token** and **OAuth2.0 Authorization Code**) are generally available and don't require contacting our CS to activate them.

As explained in the [guide](https://developers.fattureincloud.it/docs/authentication/manual-authentication), the steps to obtain a Manual Token are the following:
- Create a new App (see the [guide](https://developers.fattureincloud.it/docs/basics/create-an-app))
- Select "Token Personale" from the app page (see the [guide](https://developers.fattureincloud.it/docs/basics/authentication#-how-should-i-select-the-auth-flow-i-chose))
- Copy the Client ID from the app page
- Go to [Impostazioni > Applicazioni Collegate](https://secure.fattureincloud.it/settings-apps), click on the **Collega una nuova applicazione** button and paste the Client ID you copied before
- Select the [Scopes](https://developers.fattureincloud.it/docs/basics/scopes) 
- At the end of the procedure, you should see the Access Token on the page

Once obtained the token, you can copy and paste it into your code. You must insert it in the **Authorization** header as a token bearer (so you must prepend the "Bearer " string to the actual token (the space is not a typo).

For example:
```http
GET /c/17/entities/suppliers/16 HTTP/1.1
Host: api-v2.fattureincloud.it
Accept: application/json
Authorization: Bearer a/eyJ0eXAiOiJKV1QiL...
```

Let us know if you have any doubts about this procedure.

Please, remember that:
- this token doesn't expire and must be used safely
- if you plan to have multiple Fatture in Cloud users we don't suggest using the manual token because it is more difficult to use for the final user
- if your app is still private, then you must setup the whitelist correctly to use it with a Fatture in Cloud account; when an app is created it automatically contains the account of its owner

Thanks
Mauro

**Reply by valmoz** [reply]
If you still need help, you can check [this page](https://developers.fattureincloud.it/docs/support) or contact the support at info@fattureincloud.it 

Thanks 
Mauro

**Comment by nncodestone** [comment]
Hi Mauro,

While we are getting Access token by Manual authentication process, we are getting Client ID invalid error even though if we enter correct client ID.

May I know your availability to have a call and show you the issue?

Thanks & Regards
Nageswara Nunna​​​​
Senior Software Engineer

**Reply by valmoz** [reply, marked as answer]
Hi @nncodestone 
for our internal policies, we can only provide written support.

I checked your App, and I can see you selected OAuth 2.0 as Auth method.
<img width="627" alt="image" src="https://github.com/fattureincloud/api/assets/7254293/d04b1847-5185-410e-9d62-cc822939f7fa">

Please, switch the selected auth method and retry.

thanks
Mauro

**Reply by valmoz** [reply]
Hi @nncodestone were you able to solve your issue?

Thanks
Mauro

**Comment by nncodestone** [comment]
Hi Maura,

Yes, Thanks.


Thanks & Regards
Nageswara Nunna​​​​
Senior Software Engineer
